postman/update_env.py

```diff
@@ -0,0 +1,55 @@
+import json
+import os
+from pathlib import Path
+import sys
+
+pwd = os.getcwd()
+env_json_path = Path(f'{pwd}/postman/env.json')
+
+env = json.loads("""
+{
+	"name": "ConjurEnv",
+	"values": [
+		{
+			"key": "baseUrl",
+			"value": "http://localhost:80",
+			"enabled": true
+		},
+		{
+			"key": "account",
+			"value": "dev",
+			"enabled": true
+		},
+		{
+			"key": "default-auth",
+			"value": "authn",
+			"enabled": true
+		},
+        {
+			"key": "admin-api-key",
+			"value": "",
+			"enabled": true
+		},
+		{
+			"key": "token",
+			"value": "",
+			"enabled": true
+		},
+        {
+            "key": "formatted_token",
+            "value": "Token token=\\"{{token}}\\"",
+            "enabled": true
+        }
+	]
+}
+""")
+
+# update with command line args
+admin_api_key = sys.argv[1]
+admin_token = sys.argv[2].replace('\r','')[28:-1]
+
+env['values'][3]['value'] = admin_api_key
+env['values'][4]['value'] = admin_token
```

Comment:
does this only work for 8 minutes until the token expires? just curious - if that's so, we should probably document that

Reply:
I'm pretty sure there is some way to tell Postman how to get the API token programmatically. So you could specify to execute a `docker-compose exec cli conjur authn authenticate -H` and it would execute it before every request. The requests would take a bit longer but it would mean there was no need to re-run this script every few minutes.

Also I would be interested to know if this works with the endpoints that require using Basic auth (like rotating a users own API key). I had some issues with the Python client performing a rotation on the current user due to some authentication issues.

Reply:
My first idea was to inform the user that they would need to regenerate the token, and show them how to do it using the API through Postman, but this does seem better. I'll include it in a more detailed example.

Reply:
In Postman, you can create "Tests" in JavaScript that run after the request is completed successfully.

Here is how I use "Tests" to Base64-encode the returned session token and add it as the value of an Environment Variable located within the associated Environment Template.

```javascript
var str=btoa(responseBody)
postman.setEnvironmentVariable("dapToken", str);
```

Reply:
@infamousjoeg FYI you can send `accept-encoding` header to the authenticate endpoint with value `base64` to receive a server response of a base64-encoded token - hopefully this makes things easier for you (it's supported for Conjur v1.9.0+